### DISCUSSION TITLE
What's for the .docker directory?

### DISCUSSION THREAD

**Discussion by rrebollo** [Tecnativa/doodba-copier-template#499] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/90912322-8f2c-4be4-a7e0-32fa4c599e8d)
This directory has some .env files but I don't see any effect from these:
![image](https://github.com/user-attachments/assets/b7420ee5-06ab-48e3-a5fe-91371fbbf1bd)
env command in running container shows no evidence of this .env files been used anywhere.

**Comment by pedrobaeza** [comment]
A folder to store some secrets that are read and passed later as environment variables, like the Odoo master password, DB access password or DB creation password.

**Reply by rrebollo** [reply]
> A folder to store some secrets that are read and passed later as environment variables, like the Odoo master password, DB access password or DB creation password.

Seeing at .env files with cat I get:
![image](https://github.com/user-attachments/assets/ee53f7f7-96b3-4246-a05e-c4e411e37779)

But when echoing those vars in a temporal odoo container I get:
![image](https://github.com/user-attachments/assets/1b0d7f59-7568-4fa7-92e9-609da42b1104)

But my dooba develop environment works just fine.

I was looking for a place where to set some custom environment variable to be expanded at repos.yaml for git-aggregate task. But the only place I have found that actually works is in setup-devel.yaml environment section in the odoo service definition. Neither the container created by setup-devel,yaml o develop.yaml look like are reading this .env files. Maybe an issue?

**Reply by pedrobaeza** [reply, marked as answer]
They are not used for devel, for easing things, and if you don't put them, everything is going to work. It's intended for prod/demo environments, and for having that files populated by other systems (like Ansible).
